### PR TITLE
add trash can facade

### PR DESCRIPTION
**Pull request by itkg-alavieille**
https://github.com/open-orchestra/open-orchestra-libs/pull/15
https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/1026
https://github.com/open-orchestra/open-orchestra-model-interface/pull/108
https://github.com/open-orchestra/open-orchestra-model-bundle/pull/411

### CODE REVIEW

**Comment by itkg-alavieille** [comment]
@nicolasThal

File: ApiBundle/Controller/DeletedController.php [hunk 1]
```diff
@@ -24,15 +30,12 @@ class DeletedController extends BaseController
...
-    public function listAction()
+    public function listAction(Request $request)
     {
-        $siteId = $this->get('open_orchestra_backoffice.context_manager')->getCurrentSiteId();
-        /** @var array $nodes */
-        $nodes = $this->get('open_orchestra_model.repository.node')->findLastVersionByDeletedAndSiteId($siteId);
-        $contents = $this->get('open_orchestra_model.repository.content')->findAllDeleted();
-
-        $deleted = array_merge($nodes, $contents);
+        $mapping = $this->get('open_orchestra_base.annotation_search_reader')->extractMapping($this->getParameter('open_orchestra_model.document.trash_item.class'));
```
Comment (nicolasThal) [inline ApiBundle/Controller/DeletedController.php]: This way of getting the mapping should be done all the time, is there a ticket for that ?

**Comment by nicolasThal** [comment]
:+1:

File: ApiBundle/Controller/DeletedController.php [hunk 2]
```diff
@@ -16,7 +18,11 @@
  */
 class DeletedController extends BaseController
```
Comment (itkg-ngilain) [inline ApiBundle/Controller/DeletedController.php]: Would not it be more accurate to name this controller TrashcanController ? It seems used to display trashitems, isn't it ?

File: ApiBundle/Facade/TrashItemCollectionFacade.php [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+{
+    /**
+     * @Serializer\Type("string")
+     */
+    public $collectionName = 'trashitem';
+
+    /**
+     * @Serializer\Type("array<OpenOrchestra\ApiBundle\Facade\TrashItemFacade>")
+     */
+    protected $trashitem= array();
```
Comment (itkg-ngilain) [inline ApiBundle/Facade/TrashItemCollectionFacade.php]: checkstyle : a little space before equal sign

File: ApiBundle/Facade/TrashItemFacade.php [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+{
+    /**
+     * @Serializer\Type("string")
+     */
+    public $name;
+
+    /**
+     * @Serializer\Type("string")
+     */
+    public $deleteAt;
```
Comment (itkg-ngilain) [inline ApiBundle/Facade/TrashItemFacade.php]: deletedAt maybe ?

File: ApiBundle/Controller/DeletedController.php (hunk 1 above)
Comment (itkg-alavieille) [inline ApiBundle/Controller/DeletedController.php]: yes in the column bugs.

**Comment by itkg-alavieille** [comment]
@itkg-ngilain

File: ApiBundle/Facade/TrashItemCollectionFacade.php [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+{
+    /**
+     * @Serializer\Type("string")
+     */
+    public $collectionName = 'trashitem';
+
+    /**
+     * @Serializer\Type("array<OpenOrchestra\ApiBundle\Facade\TrashItemFacade>")
+     */
+    protected $trashitem = array();
```
Comment (nicolasThal) [inline ApiBundle/Facade/TrashItemCollectionFacade.php]: you could have put it in a plural way `trashItems`

File: ApiBundle/Facade/TrashItemCollectionFacade.php [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+/**
+ * Class TrashItemCollectionFacade
+ */
+class TrashItemCollectionFacade extends PaginateCollectionFacade
+{
+    /**
+     * @Serializer\Type("string")
+     */
+    public $collectionName = 'trashitem';
```
Comment (nicolasThal) [inline ApiBundle/Facade/TrashItemCollectionFacade.php]: modify the collection name to trash_items

File: ApiBundle/Facade/TrashItemFacade.php [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    /**
+     * @Serializer\Type("string")
+     */
+    public $name;
+
+    /**
+     * @Serializer\Type("string")
+     */
+    public $deletedAt;
+
```
Comment (nicolasThal) [inline ApiBundle/Facade/TrashItemFacade.php]: empty line to delete

File: ApiBundle/Transformer/TrashItemCollectionTransformer.php [hunk 8]
```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace OpenOrchestra\ApiBundle\Transformer;
+
+use Doctrine\Common\Collections\Collection;
+use OpenOrchestra\ApiBundle\Facade\TrashItemCollectionFacade;
+use OpenOrchestra\BaseApi\Facade\FacadeInterface;
+use OpenOrchestra\BaseApi\Transformer\AbstractTransformer;
+
+class TrashItemCollectionTransformer extends AbstractTransformer
```
Comment (nicolasThal) [inline ApiBundle/Transformer/TrashItemCollectionTransformer.php]: class comment

File: ApiBundle/Transformer/TrashItemTransformer.php [hunk 9]
```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace OpenOrchestra\ApiBundle\Transformer;
+
+use OpenOrchestra\ApiBundle\Exceptions\TransformerParameterTypeException;
+use OpenOrchestra\ApiBundle\Facade\TrashItemFacade;
+use OpenOrchestra\BaseApi\Transformer\AbstractTransformer;
+use OpenOrchestra\ModelInterface\Model\TrashItemInterface;
+
+class TrashItemTransformer extends AbstractTransformer
```
Comment (nicolasThal) [inline ApiBundle/Transformer/TrashItemTransformer.php]: class comment

File: ApiBundle/Transformer/TrashItemTransformer.php [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+    {
+        if (!$trashItem instanceof TrashItemInterface) {
+            throw new TransformerParameterTypeException();
+        }
+
+        $facade = new TrashItemFacade();
+        $facade->id = $trashItem->getId();
+        $facade->deletedAt = $trashItem->getDeletedAt();
+        $facade->name = $trashItem->getName();
+        return $facade;
```
Comment (nicolasThal) [inline ApiBundle/Transformer/TrashItemTransformer.php]: empty line before return

**Comment by itkg-alavieille** [comment]
@nicolasThal

**Comment by nicolasThal** [comment]
:+1:

**Comment by itkg-ngilain** [comment]
:+1:

**Pull request merged**